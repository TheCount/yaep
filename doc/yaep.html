<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<HTML>
<HEAD>
 <META NAME="GENERATOR" CONTENT="LinuxDoc-Tools 0.9.66">
 <TITLE>YAEP (Yet Another Earley Parser) - C interface</TITLE>
 <LINK HREF="yaep-1.html" REL=next>


</HEAD>
<BODY>
<A HREF="yaep-1.html">Next</A>
Previous
Contents
<HR>
<H1>YAEP (Yet Another Earley Parser) - C interface</H1>

<H2>Vladimir Makarov, <CODE>vmakarov@gcc.gnu.org</CODE></H2>Oct 10, 2015
<HR>
<EM>This document describes YAEP (Yet Another Earley Parser) written in C/C++.</EM>
<HR>
<P>
<H2><A NAME="toc1">1.</A> <A HREF="yaep-1.html">YAEP</A></H2>

<HR>
<A HREF="yaep-1.html">Next</A>
Previous
Contents
</BODY>
</HTML>
