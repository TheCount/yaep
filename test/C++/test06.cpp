#include<cstdio>
#include <stdlib.h>

#include"common.h"
#include "yaep.h"

static const char *input = "a+a*(a*a+a)";

static const char *description =
"\n"
"TERM;\n"
"E : T         # 0\n"
"  | E '+' T   # plus (0 2)\n"
"  ;\n"
"T : F         # 0\n"
"  | T '*' F   # mult (0 2)\n"
"  ;\n"
"F : 'a'       # 0\n"
"  | '(' E ')' # 1\n"
"  ;\n"
  ;

int
main ()
{
  yaep *e;
  struct yaep_tree_node *root;
  int ambiguous_p;

  test_input = input;
  e = new yaep ();
  if (e == NULL)
    {
      fprintf (stderr, "yaep::yaep: No memory\n");
      exit (1);
    }
  if (e->parse (test_read_token, test_syntax_error, test_parse_alloc, NULL,
		&root, &ambiguous_p))
    {
      fprintf (stderr, "yaep parse: %s\n", e->error_message ());
      exit (1);
    }
  delete e;
  exit (0);
}
