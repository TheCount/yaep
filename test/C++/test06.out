yaep::parse: undefined or bad grammar
